## Pizza Sales Exploratory Data Analysis

In [1]:
import pandas as pd
import numpy as np
import os

In [2]:
# create path to csvs
path = 'pizza_store_tables'


In [3]:
csv_list = os.listdir(path)
csv_list

['.ipynb_checkpoints',
 'orders.csv',
 'order_details - Copy.csv',
 'order_details.csv',
 'pizzas.csv',
 'pizza_types - Copy.csv',
 'pizza_types.csv']

### Orders Table EDA
- Check data types
- Check for null values
- Check for duplicates in columns that should have unique values

In [4]:
# read in orders table
order_path = os.path.join(path,'orders.csv')
order_path
orders_df = pd.read_csv(order_path)
orders_df.head()

order_id        date      time
0         1  2015-01-01  11:38:36
1         2  2015-01-01  11:57:40
2         3  2015-01-01  12:12:28
3         4  2015-01-01  12:16:31
4         5  2015-01-01  12:21:30

#### Check data types and for null values

In [5]:
orders_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21350 entries, 0 to 21349
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   order_id  21350 non-null  int64 
 1   date      21350 non-null  object
 2   time      21350 non-null  object
dtypes: int64(1), object(2)
memory usage: 500.5+ KB


No nulls, order_id is int64 data type as expected. 
Date and time are objects, may need to be converted to datetime for further analysis.

#### Check for duplicates in the order_id column

In [6]:
orders_df['order_id'].nunique()

21350

No duplicates

### Order Details table EDA
    Perform similar checks as with above table

In [7]:
# read in order_details table
order_details_path = os.path.join(path,'order_details.csv')
order_details_df = pd.read_csv(order_details_path)
order_details_df.head()

order_details_id  order_id       pizza_id  quantity
0                 1         1     hawaiian_m         1
1                 2         2  classic_dlx_m         1
2                 3         2  five_cheese_l         1
3                 4         2    ital_supr_l         1
4                 5         2     mexicana_m         1

In [8]:
order_details_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48620 entries, 0 to 48619
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   order_details_id  48620 non-null  int64 
 1   order_id          48620 non-null  int64 
 2   pizza_id          48620 non-null  object
 3   quantity          48620 non-null  int64 
dtypes: int64(3), object(1)
memory usage: 1.5+ MB


No null values

In [9]:
order_details_df.nunique()

order_details_id    48620
order_id            21350
pizza_id               91
quantity                4
dtype: int64

#### Results:
    order_details_id: no duplicates, this is the primary key for this table
    order_id:21350 unique values corresponds correctly to order_id of orders table; this is foreign key
    pizza_id: 91 unique values, when checking pizzas table, should not be fewer than 91 pizza_ids in that table

#### Check min and max order quantities to ensure no odd results such as negative units ordered

In [10]:
order_details_df['quantity'].max()


4

In [11]:
order_details_df['quantity'].min()

1

### Pizzas table EDA

In [12]:
# read in pizzas csv
pizzas_path = os.path.join(path,'pizzas.csv')
pizzas_df = pd.read_csv(pizzas_path)
pizzas_df.head()

pizza_id pizza_type_id size  price
0   bbq_ckn_s       bbq_ckn    S  12.75
1   bbq_ckn_m       bbq_ckn    M  16.75
2   bbq_ckn_l       bbq_ckn    L  20.75
3  cali_ckn_s      cali_ckn    S  12.75
4  cali_ckn_m      cali_ckn    M  16.75

There are separate fields pizza_id and pizza_type_id, where pizza_id is a combination of pizza_type and size, so there should be far more pizza_ids than pizza_type_ids

In [13]:
pizzas_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96 entries, 0 to 95
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   pizza_id       96 non-null     object 
 1   pizza_type_id  96 non-null     object 
 2   size           96 non-null     object 
 3   price          96 non-null     float64
dtypes: float64(1), object(3)
memory usage: 3.1+ KB


No null values, price is float64 as expected

In [14]:
pizzas_df.describe()

price
count  96.000000
mean   16.440625
std     4.090266
min     9.750000
25%    12.500000
50%    16.250000
75%    20.250000
max    35.950000

In [15]:
pizzas_df['pizza_id'].nunique()

96

There 96 unique pizza_ids, which validates the 91 from the order_details_table, and 5 pizza_ids have not been ordered in the year

In [16]:
pizzas_df['pizza_type_id'].nunique()

32

There are 32 unique pizza_type_ids, so should not be fewer than 32 in the pizza_types_table

### Pizza Types EDA

In [17]:
pizza_types_df = pd.read_csv(os.path.join(path,'pizza_types.csv'),encoding= 'unicode_escape')
pizza_types_df.head()

pizza_type_id                          name category  \
0       bbq_ckn    The Barbecue Chicken Pizza  Chicken   
1      cali_ckn  The California Chicken Pizza  Chicken   
2   ckn_alfredo     The Chicken Alfredo Pizza  Chicken   
3     ckn_pesto       The Chicken Pesto Pizza  Chicken   
4    southw_ckn   The Southwest Chicken Pizza  Chicken   

                                         ingredients  
0  Barbecued Chicken, Red Peppers, Green Peppers,...  
1  Chicken, Artichoke, Spinach, Garlic, Jalapeno ...  
2  Chicken, Red Onions, Red Peppers, Mushrooms, A...  
3  Chicken, Tomatoes, Red Peppers, Spinach, Garli...  
4  Chicken, Tomatoes, Red Peppers, Red Onions, Ja...

In [18]:
pizza_types_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   pizza_type_id  32 non-null     object
 1   name           32 non-null     object
 2   category       32 non-null     object
 3   ingredients    32 non-null     object
dtypes: object(4)
memory usage: 1.1+ KB


No values

In [19]:
pizza_types_df.nunique()

pizza_type_id    32
name             32
category          4
ingredients      32
dtype: int64

There are 32 unique pizza_type_ids with no duplicates, which validates the 32 from the pizzas table. There are 32 unique sets of ingredients, which confirms there are no duplicated ingredient sets for the 32 pizza types.